src/main/java/edu/ie3/datamodel/io/sink/CsvFileSink.java

```diff
@@ -246,7 +248,10 @@ public <E extends TimeSeriesEntry<V>, V extends Value> void persistTimeSeries(
     try {
       TimeSeriesProcessorKey key = new TimeSeriesProcessorKey(timeSeries);
       String[] headerElements = csvHeaderElements(processorProvider.getHeaderElements(key));
-      BufferedCsvWriter writer = connector.getOrInitWriter(timeSeries, headerElements, csvSep);
+      BufferedCsvWriter writer =
+          connector.getOrInitWriter(
+              timeSeries,
+              () -> new CsvFileDefinition(timeSeries, headerElements, csvSep, fileNamingStrategy));
```

Comment:
What benefit does passing the CsvFileDefinition as a supplier have? From far, it just seems like this makes the code a bit more complicated than it has to be.